src/brainfuck/Metrics.java

```diff
@@ -0,0 +1,72 @@
+package brainfuck;
+
+/**
+ * Define the different Metrics for the current instance of the program.
+ *
+ * @author Julien Lemaire
+ */
+public enum Metrics {
+	PROG_SIZE, EXEC_TIME, EXEC_MOVE, DATA_MOVE, DATA_WRITE, DATA_READ;
+
+	/**
+	 * The value of the current Metric.
+	 */
+	private long value;
+
+	/**
+	 * Storing time after call of start method.
+	 */
+	private long time;
+
+	/**
+	 * Increment the value by one.
+	 */
+	public void incr() {
+		this.value++;
+	}
+
+	/**
+	 * Set the value of the metric with a new long value.
+	 *
+	 * @param value the new value of the metrics.
+	 */
+	public void set(long value) {
+		this.value = value;
+	}
+
+	/**
+	 * Start a timer.
+	 */
+	public void start() {
+		this.time = System.currentTimeMillis();
```

Comment:
I turned it that way to avoid getting first result of `System.currentTimeMillis()` when using `value()` method after `start()` and before `stop()`. But this is not a normal use of `EXEC_TIME` anyway, so I suppose you're right :).